lang/lpython/Portfile

```diff
@@ -0,0 +1,78 @@
+PortGroup               github 1.0
+
+github.setup            lcompilers lpython 0.20.0 v
+revision                0
+categories              lang python
+maintainers             {@barracuda156 gmail.com:vital.had} openmaintainer
+license                 BSD
+description             High-performance typed Python compiler
+long_description        LPython aggressively optimizes type-annotated Python code. \
+                        It has several backends, including LLVM, C, C++ and WASM. \
+                        LPython’s primary tenet is speed.
+homepage                https://lpython.org
+
+checksums               rmd160  01cdab5767af01d017339f79355c34e9ab2ec8d7 \
+                        sha256  741fb64ddf70528eb713103d5b6e2068344b5768b546b0c21145d5a1507635a0 \
+                        size    1196647
+github.tarball_from     archive
+
+set port_libfmt         libfmt10
+cmake.module_path-append \
+                        ${prefix}/lib/${port_libfmt}/cmake
+
+depends_build-append    port:bash \
+                        port:bison \
+                        port:re2c
+depends_lib-append      path:lib/pkgconfig/RapidJSON.pc:rapidjson \
+                        port:zlib \
+                        port:${port_libfmt}
+
+set py_ver              3.11
+set py_ver_nodot        [string map {. {}} ${py_ver}]
+depends_lib-append      port:python${py_ver_nodot}
+
+patchfiles-append       0001-stacktrace.cpp-include-syslimits.h-for-PATH_MAX.patch \
+                        0002-UserOverride.cmake-fix-native-optflag-for-PowerPC.patch \
+                        0003-Install-libs-into-a-sane-location.patch \
+                        0004-parser_stype.h-allow-building-for-ppc.patch \
+                        0005-patch-build0.sh.patch \
+                        0006-patch-version.patch
+
+configure.python        ${prefix}/bin/python${py_ver}
+
+post-patch {
+    reinplace "s|@VERSION@|${version}|" ${worksrcpath}/ci/version.sh
+    reinplace "s|@PYTHON@|${configure.python}|g" ${worksrcpath}/build0.sh
+    reinplace "s|/usr/bin/env python|${configure.python}|" ${worksrcpath}/run_tests.py
+}
+
+compiler.cxx_standard   2017
+
+# TODO: add support for LLVM for x86 and aarch64, as it is done for lfortran,
+# once it gets fixed: https://github.com/lcompilers/lpython/issues/2462
+
+configure.args-append   -DLPYTHON_BUILD_ALL=YES \
+                        -DLPYTHON_STATIC_BIN=NO \
+                        -DWITH_FMT=YES \
+                        -DWITH_INTRINSIC_MODULES=NO \
+                        -DWITH_JSON=YES \
+                        -DWITH_LCOMPILERS_FAST_ALLOC=YES \
+                        -DWITH_LLVM=NO \
+                        -DWITH_MACHO=YES \
+                        -DWITH_RUNTIME_LIBRARY=YES \
+                        -DWITH_STACKTRACE=NO \
+                        -DWITH_UNWIND=NO \
+                        -DWITH_ZLIB=YES
+
+pre-configure {
+    system -W ${worksrcpath} "sh ./build0.sh"
+}
+
+cmake.build_type        Release
```

Comment:
can this be left to the default, which is (I think) "macports" and sets correct flags and such?